File: kk.c
/*

70940507 and 60940371

CS124 Coding Assignment 3

Usage: ./kk infile

*/

#include <time.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(int argc, char* argv[]){
	if (argc != 2){
        printf("usage: ./kk infile\n");
        return -1;
    }

    char* infile = argv[1];

    char command[100];
	snprintf(command, sizeof(command), "python kk.py %s", infile);

    system(command);
}

File: kk.py
## CS124 Programming Assignment 3
## HUIDs: 70940507 and 60940371
## random_kk.py

from bisect import insort_left
import numpy as np
from copy import copy

import sys

if len(sys.argv) != 2:
	print("usage: python kk.py infile")
	quit()

infile = sys.argv[1]

A = np.loadtxt(infile).tolist()

def KK(A_p): # O(n*n), so not what we want
	A = copy(A_p)
	A.sort() # O(n log n)
	n = len(A)
	while n > 1: # O(n) times to insort * O(n) insort
		delta = abs(A.pop() - A.pop()) 
		insort_left(A, delta)
		n -= 1
	return A[0]

print(KK(A))